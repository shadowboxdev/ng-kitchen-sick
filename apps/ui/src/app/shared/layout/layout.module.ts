import { CommonModule } from '@angular/common';
import { NgModule, Type } from '@angular/core';

import { LayoutModule as CdkLayoutModule } from '@angular/cdk/layout';
import { CdkMenuModule } from '@angular/cdk/menu';
import { MatButtonModule } from '@angular/material/button';
import { MatCardModule } from '@angular/material/card';
import { MatIconModule } from '@angular/material/icon';
import { MatListModule } from '@angular/material/list';
import { MatSidenavModule } from '@angular/material/sidenav';
import { MatToolbarModule } from '@angular/material/toolbar';

import {
  FaIconLibrary,
  FontAwesomeModule
} from '@fortawesome/angular-fontawesome';
import {
  faGithub,
  faMediumM,
  faTwitter,
  faInstagram,
  faYoutube
} from '@fortawesome/free-brands-svg-icons';
import {
  faCog,
  faBars,
  faRocket,
  faPowerOff,
  faUserCircle,
  faPlayCircle
} from '@fortawesome/free-solid-svg-icons';

import { LayoutComponent } from './components/layout.component';
import { ToolbarComponent } from './components/toolbar.component';
import { OutlinedIconDirective } from './directives/outlined-icon.directive';

const NG_IMPORTS: Type<unknown>[] = [CommonModule];

const MAT_IMPORTS: Type<unknown>[] = [
  CdkMenuModule,
  CdkLayoutModule,
  MatSidenavModule,
  MatToolbarModule,
  MatIconModule,
  MatButtonModule,
  MatListModule,
  MatCardModule
];

const OTHER_IMPORTS: Type<unknown>[] = [FontAwesomeModule];

const DECLARATIONS: Type<unknown>[] = [
  LayoutComponent,
  ToolbarComponent,
  OutlinedIconDirective
];

@NgModule({
  declarations: [...DECLARATIONS],
  imports: [...NG_IMPORTS, ...MAT_IMPORTS, ...OTHER_IMPORTS],
  exports: [...DECLARATIONS],
  providers: []
})
export class LayoutModule {
  constructor(private readonly _faIconLibrary: FaIconLibrary) {
    this._addFaIcons();
  }

  private _addFaIcons(): void {
    this._faIconLibrary.addIcons(
      faCog,
      faBars,
      faRocket,
      faPowerOff,
      faUserCircle,
      faPlayCircle,
      faGithub,
      faMediumM,
      faTwitter,
      faInstagram,
      faYoutube
    );
  }
}
