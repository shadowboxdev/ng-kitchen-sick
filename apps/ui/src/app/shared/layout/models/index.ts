export * from './layout-container.interface';
