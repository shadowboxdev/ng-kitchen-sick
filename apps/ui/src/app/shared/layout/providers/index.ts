export * from './layout-container.provider';
