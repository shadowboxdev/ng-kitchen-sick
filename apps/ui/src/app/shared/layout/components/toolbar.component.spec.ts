import { Spectator, createComponentFactory } from '@ngneat/spectator/jest';

import { LayoutModule } from '../layout.module';
import { ToolbarComponent } from './toolbar.component';

describe('ToolbarComponent', () => {
  let spectator: Spectator<ToolbarComponent>;

  const createComponent = createComponentFactory({
    component: ToolbarComponent,
    imports: [LayoutModule]
  });

  beforeEach(() => (spectator = createComponent()));

  it('should create component', () => {
    expect(spectator.component).toBeTruthy();
  });

  it('should render dashboard', () => {
    expect(spectator.fixture).toMatchSnapshot();
  });
});
