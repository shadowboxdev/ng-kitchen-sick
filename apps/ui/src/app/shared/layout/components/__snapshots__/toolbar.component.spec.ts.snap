// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolbarComponent should render dashboard 1`] = `
<sdw-toolbar
  __ngContext__={[Function Number]}
  _container="null"
  _positions={[Function Array]}
  folders={[Function Array]}
  notes={[Function Array]}
>
  <mat-toolbar
    class="mat-toolbar sm:fized sm:z-2 flex flex-col mat-primary mat-toolbar-single-row"
    color="primary"
  >
    <button
      class="mat-focus-indicator mat-icon-button mat-button-base _mat-animation-noopable"
      mat-icon-button=""
    >
      <span
        class="mat-button-wrapper"
      >
        <mat-icon
          aria-hidden="true"
          class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color"
          data-mat-icon-type="font"
          role="img"
        >
          menu
        </mat-icon>
      </span>
      <span
        class="mat-ripple mat-button-ripple mat-button-ripple-round"
        matripple=""
      />
      <span
        class="mat-button-focus-overlay"
      />
    </button>
    <h1
      class="ml-8 hidden sm:block"
    >
      Responsive App
    </h1>
    <span
      class="flex flex-1"
    />
    <div
      class="w-72 text-black"
    >
      <div
        class="relative"
      >
        <input
          autocomplete="off"
          class="w-full rounded-lg border border-gray-200"
          placeholder="Search..."
          type="search"
        />
      </div>
    </div>
    <div
      class="hidden md:block"
    >
      <div
        class="ml-4 flex items-center md:ml-6"
      >
        <button
          class="mat-focus-indicator max-w-xs rounded-full flex items-center text-sm focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-800 focus:ring-white mat-icon-button mat-button-base _mat-animation-noopable"
          mat-icon-button=""
        >
          <span
            class="mat-button-wrapper"
          >
            <span
              class="sr-only"
            >
              View notifications
            </span>
            <mat-icon
              aria-hidden="true"
              class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color material-icons-outlined"
              data-mat-icon-type="font"
              outlined=""
              role="img"
            >
              notifications
            </mat-icon>
          </span>
          <span
            class="mat-ripple mat-button-ripple mat-button-ripple-round"
            matripple=""
          />
          <span
            class="mat-button-focus-overlay"
          />
        </button>
        <div
          class="ml-3 relative"
        >
          <button
            aria-expanded="false"
            aria-haspopup="true"
            class="cdk-menu-trigger max-w-xs bg-gray-800 rounded-full flex items-center text-sm focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-800 focus:ring-white"
            data-cdk-menu-stack-id="1"
            id="user-menu-button"
            role="button"
          >
            <span
              class="sr-only"
            >
              Open user menu
            </span>
            <img
              alt=""
              class="h-8 w-8 rounded-full"
              src="https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80"
            />
          </button>
        </div>
      </div>
    </div>
  </mat-toolbar>
</sdw-toolbar>
`;
